### DISCUSSION TITLE
Mesh_3/mesh_polyhedral_domain_with_surface_inside

### DISCUSSION THREAD

**Discussion by KALEEMULLAHJANKHAN** [CGAL/cgal#6351] [General]
For what purpose we can mainly use this code "[Mesh_3/mesh_polyhedral_domain_with_surface_inside](https://doc.cgal.org/latest/Mesh_3/Mesh_3_2mesh_polyhedral_domain_with_surface_inside_8cpp-example.html)"? I did not find any detailed applications and use of this code. 
Brief explanation with example would be appreciated.
Thanks and kind regards

**Comment by sloriot** [comment]
It generates a tetrahedral mesh of some surfaces with boundaries within a given domain (here the domain is a rectangular hexahedron).
The input surfaces will be remeshed and be represented by some faces in the tetrahedral mesh.

**Reply by KALEEMULLAHJANKHAN** [reply]
Thanks for the response. I am clear with the explanation as you have provided. I actually confuse about the mechanism; for example, what can be the internal domain. Can we put several polyhedrons inside a single domain (as in the given hexahedron)?

**Comment by lrineau** [comment]
In addition what Sébastien explained, that example is documented in the User Manual _CGAL 5.4 - 3D Mesh Generation_, section _4.3.2 Remeshing a Polyhedral Domain with Surfaces_: https://doc.cgal.org/5.4/Mesh_3/index.html#title23

**Reply by KALEEMULLAHJANKHAN** [reply]
Thank you sir for the response. The given explanation regarding the code is not much enough, I came across with the documentation but did not get any clear idea. In my case; I have a domain composed of more than 100 polyhedral blocks, now I want to individually mesh all those blocks in a single domain.

**Reply by lrineau** [reply]
The example `mesh_polyhedral_domain_with_surface_inside.cpp` uses two AABB trees of triangles:
  - one, built from multiple non-intersecting polyhedra, is used to decide which triangles of the triangulation are associated to the surface mesh,
  - the other one is buil tfrom  only one polyhedron, assumed to be closed, and is used to decide which tetrahedra of the triangulation are associated to the volumic mesh.

**Reply by KALEEMULLAHJANKHAN** [reply]
![One fracture vtk](https://user-images.githubusercontent.com/89472468/156864890-f8596834-119e-4bfd-a949-639b3cd95680.JPG)
Dear sir, Can I embed the attached polygon into a cube using this code (CGAL 5.4 - 3D Mesh Generation, section 4.3.2 Remeshing a Polyhedral Domain with Surfaces: https://doc.cgal.org/5.4/Mesh_3/index.html#title23)? The polygonal structure is attached here. Thanks

**Reply by lrineau** [reply]
Yes, you can use that code example to mesh the polygon.

Note that both the polygon and the enclosing cube must be triangulated, so that the CGAL mesh generation can work with them.
